tests/ui/type-alias-impl-trait/coherence.rs

```diff
@@ -12,6 +12,6 @@ fn use_alias<T>(val: T) -> AliasOfForeignType<T> {
 }
 
 impl<T> foreign_crate::ForeignTrait for AliasOfForeignType<T> {}
```

Comment:
remove the type parameter or use a primitive type instead.
```suggestion
impl foreign_crate::ForeignTrait for AliasOfForeignType<()> {}
```